mlrun/db/httpdb.py

```diff
@@ -1862,37 +1866,72 @@ def create_project_secrets(
     def get_project_secrets(
         self,
         project: str,
-        token: str,
+        token: str = None,
         provider: Union[
             str, schemas.SecretProviderName
         ] = schemas.SecretProviderName.vault,
         secrets: List[str] = None,
     ) -> schemas.SecretsData:
-        """ Retrieve project-context secrets from Vault.
+        """ Retrieve project-context secrets from Vault or Kubernetes.
 
         Note:
-                This method is currently in technical preview, and requires a HashiCorp Vault infrastructure
-                properly set up and connected to the MLRun API server.
+                This method for Vault functionality is currently in technical preview, and requires a HashiCorp Vault
+                infrastructure properly set up and connected to the MLRun API server.
 
         :param project: The project name.
-        :param token: Vault token to use for retrieving secrets. Must be a valid Vault token, with permissions to
-            retrieve secrets of the project in question.
-        :param provider: The name of the secrets-provider to work with. Currently only ``vault`` is supported.
+        :param token: Vault token to use for retrieving secrets. Use only if ``provider`` is ``vault``.
+            Must be a valid Vault token, with permissions to retrieve secrets of the project in question.
+        :param provider: The name of the secrets-provider to work with. Either ``vault`` or ``kubernetes``.
         :param secrets: A list of secret names to retrieve. An empty list ``[]`` will retrieve all secrets assigned
-            to this specific project.
+            to this specific project. ``kubernetes`` provider only supports an empty list.
```

Comment:
Why not adding an option to the API to return only specific secrets ?

Reply:
Since the API returns only the secret keys, there isn't a lot of value in providing a list of keys, as it will simply return the list of keys from the input that exist in the k8s secret. Also, the get API is expected to be used mostly by the UI which doesn't need a partial list filter.